### PR TITLE
Additional runtime tests for the test-runtime

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: substrate/ed25519/src/lib.rs [hunk 1]
```diff
@@ -59,6 +58,23 @@ impl Public {
 		r.copy_from_slice(data);
 		Public(r)
 	}
+
+	/// Return a `Vec<u8>` filled with raw data.
+	pub fn to_raw_vec(&self) -> Vec<u8> {
```
Comment (rphmeier) [inline substrate/ed25519/src/lib.rs]: `to_x` typically implies a consuming conversion. (whereas `as_x` implies a cheap conversion).

File: substrate/ed25519/src/lib.rs [hunk 2]
```diff
@@ -80,31 +96,32 @@ impl Pair {
...
 	}
-	/// Make a new key pair from the raw secret.
-	pub fn from_secret(secret: &[u8; 32]) -> Pair {
-		let mut pkcs8_bytes: Vec<_> = FromHex::from_hex("302e020100300506032b657004220420").unwrap();
-		pkcs8_bytes.extend_from_slice(&secret[..]);
-		Pair(signature::Ed25519KeyPair::from_pkcs8_maybe_unchecked(untrusted::Input::from(&pkcs8_bytes)).unwrap())
-	}
+
+	/// Synonym for `from_seed`. @deprecated
+	pub fn from_secret(seed: &[u8; 32]) -> Pair { Self::from_seed(seed) }
```
Comment (rphmeier) [inline substrate/ed25519/src/lib.rs]: `#[deprecated]`
https://stackoverflow.com/a/37766834/1745195

but given that it's not actually released we could just remove?
it also does seem useful; what's the rationale for deprecation?

File: substrate/runtime-support/src/lib.rs [hunk 3]
```diff
@@ -22,18 +22,17 @@ extern crate substrate_runtime_std as rstd;
...
 #[cfg(feature = "std")]
-mod testing;
+mod keyring;
 
 pub use self::storage::StorageVec;
 pub use self::hashable::Hashable;
 
 #[cfg(feature = "std")]
-pub use self::testing::{one, two};
+pub use self::keyring::Keyring;
```
Comment (rphmeier) [inline substrate/runtime-support/src/lib.rs]: when this is re-exported, it's not clear that it's for testing.

Comment (gavofyork) [reply substrate/runtime-support/src/lib.rs]: well... could export only with testing config, i guess..

Comment (gavofyork) [reply substrate/runtime-support/src/lib.rs]: actually that won't work

Comment (rphmeier) [reply substrate/runtime-support/src/lib.rs]: yeah, sadly compiling with `cfg(test)` doesn't carry down to the dependencies. Why not extract it out to a separate library? I don't see any reason it should live in `runtime-support` when it's more of a testing utilities kind of thing.

File: polkadot/runtime/Cargo.toml [hunk 4]
```diff
@@ -7,6 +7,7 @@ authors = ["Parity Technologies <admin@parity.io>"]
 rustc-hex = "1.0"
 hex-literal = "0.1.0"
 log = { version = "0.3", optional = true }
+substrate-keyring = { path = "../../substrate/keyring", optional = true }
```
Comment (rphmeier) [inline polkadot/runtime/Cargo.toml]: when only used for tests it can go in the `dev-dependencies` section

File: polkadot/executor/Cargo.toml [hunk 5]
```diff
@@ -13,6 +13,7 @@ substrate-runtime-io = { path = "../../substrate/runtime-io" }
 substrate-runtime-support = { path = "../../substrate/runtime-support" }
 substrate-state-machine = { path = "../../substrate/state-machine" }
 substrate-executor = { path = "../../substrate/executor" }
+substrate-keyring = { path = "../../substrate/keyring" }
```
Comment (rphmeier) [inline polkadot/executor/Cargo.toml]: same here

File: substrate/executor/src/native_executor.rs [hunk 6]
```diff
@@ -69,3 +70,37 @@ impl<D: NativeExecutionDispatch + Sync + Send> CodeExecutor for NativeExecutor<D
...
 }
+
+#[macro_export]
+macro_rules! native_executor_instance {
+	(pub $name:ident, $dispatcher:path, $code:expr) => {
+		pub struct $name;
+		native_executor_instance!(IMPL $name, $dispatcher, $code);
+	};
+	($name:ident, $dispatcher:path, $code:expr) => {
+		/// A null struct which implements `NativeExecutionDispatch` feeding in the hard-coded runtime.
```
Comment (rphmeier) [inline substrate/executor/src/native_executor.rs]: not a huge fan of the term "null struct". more accurate is "unit struct"

**Comment by gavofyork** [comment]
@rphmeier ^^^

**Pull request merged**